service/history/workflow/mutable_state_impl.go

```diff
@@ -4403,6 +4410,12 @@ func (_ *MutableStateImpl) unixNanoToTime(
 	return time.Unix(0, timestampNanos).UTC()
 }
 
+func (e *MutableStateImpl) setNewExecutionRunID(newExecutionRunID string) {
+	e.executionInfo.NewExecutionRunId = newExecutionRunID
+	// Report to parent if there is no new execution, i.e. no more retries or cron or continue-as-new.
+	e.executionInfo.ReportCompletionToParent = newExecutionRunID == ""
```

Comment:
I prefer not to have this extra field and rely on NewExecutionRunId to tell if you need to report to parent or not.

Reply:
I agree, I'll remove it